src/app/settings/admin/constraint-templates/component.ts

```diff
@@ -0,0 +1,174 @@
+    this.dataSource.data = this.constraintTemplates;
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  isPaginatorVisible(): boolean {
+    return (
+      this.constraintTemplates &&
+      this.constraintTemplates.length > 0 &&
+      this.paginator &&
+      this.constraintTemplates.length > this.paginator.pageSize
+    );
+  }
+
+  loadConstraintTemplates(): void {
+    this._apiService
+      .getConstraintTemplates()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(constraintTemplates => {
+        this.constraintTemplates = constraintTemplates;
+        this.dataSource.data = this.constraintTemplates;
+      });
+  }
+
+  add(): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Add Constraint Template',
+        isEditing: false,
+        confirmLabel: 'Add',
+      },
+    };
+
+    this._matDialog
+      .open(ConstraintTemplatesDataDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(constraintTemplate => !!constraintTemplate))
+      .pipe(take(1))
+      .subscribe((result: ConstraintTemplate) => {
+        this._add(result);
+        this.loadConstraintTemplates();
+      });
+  }
+
+  private _add(constraintTemplate: ConstraintTemplate): void {
+    this._apiService
+      .createConstraintTemplate(constraintTemplate)
+      .pipe(take(1))
+      .subscribe(constraintTemplate => {
+        this._notificationService.success(`The constraint template ${constraintTemplate.name} was created`);
+      });
+  }
+
+  edit(constraintTemplate: ConstraintTemplate): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Edit Constraint Template',
+        constraintTemplate: constraintTemplate,
+        isEditing: true,
+        confirmLabel: 'Edit',
+      },
+    };
+
+    this._matDialog
+      .open(ConstraintTemplatesDataDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(constraintTemplate => !!constraintTemplate))
+      .pipe(take(1))
+      .subscribe((result: ConstraintTemplate) => {
+        this._edit(constraintTemplate, result);
+      });
+  }
+
+  private _edit(original: ConstraintTemplate, edited: ConstraintTemplate): void {
+    this._apiService
+      .patchConstraintTemplate(original.name, edited)
+      .pipe(take(1))
+      .subscribe(constraintTemplate => {
+        this._notificationService.success(`The constraint template ${constraintTemplate.name} was updated`);
+      });
+  }
+
+  delete(constraintTemplate: ConstraintTemplate): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Delete Constraint Template',
+        message: `Are you sure you want to delete the constraint template ${constraintTemplate.name}?
+                  <p class="km-confirmation-dialog-delete-warning"> <i class="km-icon-warning"></i> Deleting this constraint template will cause all constraints related to it to be deleted as well.</p>`,
```

Comment:
Please add a specific component with the template for this. Having HTML code in typescript is quite confusing and harder to maintain. Even if it seems simple, it still requires adding styling inside the main style file as there is no way to use a dedicated style file here.